test_reorg: sometimes we enter the test with "self.ledger.headers.height" at 205 instead of expected 206

I think it's a problem with `generate_and_wait()` used during test setup. It waits on a TX, but not for the target height (206).

https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true

```
203
======================================================================
[204](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:205)
FAIL: test_reorg (integration.takeovers.test_resolve_command.ResolveAfterReorg)
[205](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:206)
----------------------------------------------------------------------
[206](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:207)
Traceback (most recent call last):
[207](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:208)
  File "/home/runner/work/lbry-sdk/lbry-sdk/lbry/testcase.py", line 145, in run
[208](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:209)
    self.loop.run_until_complete(maybe_coroutine)
[209](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:210)
  File "/opt/hostedtoolcache/Python/3.7.13/x64/lib/python3.7/asyncio/base_events.py", line 587, in run_until_complete
[210](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:211)
    return future.result()
[211](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:212)
  File "/home/runner/work/lbry-sdk/lbry-sdk/tests/integration/takeovers/test_resolve_command.py", line 1573, in test_reorg
[212](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:213)
    self.assertEqual(self.ledger.headers.height, 206)
[213](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:214)
AssertionError: 205 != 206
[214](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:215)

[215](https://github.com/moodyjon/lbry-sdk/runs/8270128792?check_suite_focus=true#step:11:216)
----------------------------------------------------------------------
```

--- a/lbry/testcase.py
+++ b/lbry/testcase.py
@@ -298,13 +298,15 @@ class IntegrationTestCase(AsyncioTestCase):
 
     async def generate_and_wait(self, blocks_to_generate, txids, ledger=None):
         if blocks_to_generate > 0:
-            watcher = (ledger or self.ledger).on_transaction.where(
+            watcher1 = (ledger or self.ledger).on_transaction.where(
                 lambda e: ((e.tx.id in txids and txids.remove(e.tx.id)), len(txids) <= 0)[-1]  # multi-statement lambda
             )
+            watcher2 = (ledger or self.ledger).on_header.where(self.blockchain.is_expected_block)
             self.conductor.spv_node.server.synchronized.clear()
             await self.blockchain.generate(blocks_to_generate)
             height = self.blockchain.block_expected
-            await watcher
+            await watcher1
+            await watcher2
             while True:
                 await self.conductor.spv_node.server.synchronized.wait()
                 self.conductor.spv_node.server.synchronized.clear()

--- a/lbry/testcase.py
+++ b/lbry/testcase.py
@@ -298,23 +298,11 @@ class IntegrationTestCase(AsyncioTestCase):
 
     async def generate_and_wait(self, blocks_to_generate, txids, ledger=None):
         if blocks_to_generate > 0:
-            watcher1 = (ledger or self.ledger).on_transaction.where(
+            watcher = (ledger or self.ledger).on_transaction.where(
                 lambda e: ((e.tx.id in txids and txids.remove(e.tx.id)), len(txids) <= 0)[-1]  # multi-statement lambda
             )
-            watcher2 = (ledger or self.ledger).on_header.where(self.blockchain.is_expected_block)
-            self.conductor.spv_node.server.synchronized.clear()
-            await self.blockchain.generate(blocks_to_generate)
-            height = self.blockchain.block_expected
-            await watcher1
-            await watcher2
-            while True:
-                await self.conductor.spv_node.server.synchronized.wait()
-                self.conductor.spv_node.server.synchronized.clear()
-                if self.conductor.spv_node.server.db.db_height < height:
-                    continue
-                if self.conductor.spv_node.server._es_height < height:
-                    continue
-                break
+            await self.generate(blocks_to_generate)
+            await watcher
 
     def on_address_update(self, address):
         return self.ledger.on_transaction.where(

--- a/lbry/wallet/wallet.py
+++ b/lbry/wallet/wallet.py
@@ -177,6 +177,8 @@ class Wallet:
         except zlib.error as e:
             if "incorrect header check" in e.args[0].lower():
                 raise InvalidPasswordError()
+            if "unknown compression method" in e.args[0].lower():
+                raise InvalidPasswordError()
             raise
         return json.loads(decompressed)
 

--- a/lbry/wallet/wallet.py
+++ b/lbry/wallet/wallet.py
@@ -10,6 +10,7 @@ from collections import UserDict
 from hashlib import sha256
 from operator import attrgetter
 from lbry.crypto.crypt import better_aes_encrypt, better_aes_decrypt
+from lbry.error import InvalidPasswordError
 from .account import Account
 
 if typing.TYPE_CHECKING:
@@ -171,7 +172,12 @@ class Wallet:
     @classmethod
     def unpack(cls, password, encrypted):
         decrypted = better_aes_decrypt(password, encrypted)
-        decompressed = zlib.decompress(decrypted)
+        try:
+            decompressed = zlib.decompress(decrypted)
+        except zlib.error as e:
+            if "incorrect header check" in e.args[0].lower():
+                raise InvalidPasswordError()
+            raise
         return json.loads(decompressed)
 
     def merge(self, manager: 'WalletManager',

--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(
             'jsonschema==4.4.0',
         ],
         'hub': [
-            'hub@git+https://github.com/lbryio/hub.git@75d53c3c6fe2916aa42ce583e9f9ea9347ab7a82'
+            'hub@git+https://github.com/lbryio/hub.git@024aceda53fe6d1ab8d519b73584437c25de6975'
         ]
     },
     classifiers=[

--- a/setup.py
+++ b/setup.py
@@ -57,7 +57,7 @@ setup(
             'jsonschema==4.4.0',
         ],
         'hub': [
-            'hub@git+https://github.com/lbryio/hub.git@9e3963ba23abf2f9ace4654089a7df1fd553f121'
+            'hub@git+https://github.com/lbryio/hub.git@75d53c3c6fe2916aa42ce583e9f9ea9347ab7a82'
         ]
     },
     classifiers=[

--- a/tests/integration/blockchain/test_wallet_commands.py
+++ b/tests/integration/blockchain/test_wallet_commands.py
@@ -22,6 +22,7 @@ class WalletCommands(CommandTestCase):
 
     async def test_wallet_syncing_status(self):
         address = await self.daemon.jsonrpc_address_unused()
+        await self.ledger._update_tasks.done.wait()
         self.assertFalse(self.daemon.jsonrpc_wallet_status()['is_syncing'])
         await self.send_to_address_and_wait(address, 1)
         await self.ledger._update_tasks.started.wait()

--- a/tests/integration/blockchain/test_wallet_commands.py
+++ b/tests/integration/blockchain/test_wallet_commands.py
@@ -1,6 +1,8 @@
 import asyncio
 import json
+import string
 from binascii import unhexlify
+from random import Random
 
 from lbry.wallet import ENCRYPT_ON_DISK
 from lbry.error import InvalidPasswordError
@@ -385,9 +387,16 @@ class WalletEncryptionAndSynchronization(CommandTestCase):
         data = await daemon2.jsonrpc_sync_apply('password2')
         # sync_apply doesn't save password if encrypt-on-disk is False
         self.assertEqual(wallet2.encryption_password, None)
-        # need to use new password2 in sync_apply
-        with self.assertRaises(InvalidPasswordError):
-            await daemon.jsonrpc_sync_apply('password', data=data['data'], blocking=True)
+
+        # Need to use new password2 in sync_apply. Attempts with other passwords
+        # should fail consistently with InvalidPasswordError.
+        random = Random('password')
+        for i in range(200):
+            bad_guess = ''.join(random.choices(string.digits + string.ascii_letters + string.punctuation, k=40))
+            self.assertNotEqual(bad_guess, 'password2')
+            with self.assertRaises(InvalidPasswordError):
+                await daemon.jsonrpc_sync_apply(bad_guess, data=data['data'], blocking=True)
+
         await daemon.jsonrpc_sync_apply('password2', data=data['data'], blocking=True)
         # sync_apply with new password2 also sets it as new local password
         self.assertEqual(wallet.encryption_password, 'password2')

--- a/tests/integration/claims/test_claim_commands.py
+++ b/tests/integration/claims/test_claim_commands.py
@@ -1597,7 +1597,7 @@ class StreamCommands(ClaimTestCase):
         # blocked content is not resolveable
         error = (await self.resolve('lbry://@some_channel/bad_content'))['error']
         self.assertEqual(error['name'], 'BLOCKED')
-        self.assertTrue(error['text'].startswith(f"Resolve of 'lbry://@some_channel#{some_channel_id[:1]}/bad_content#{bad_content_id[:1]}' was censored"))
+        self.assertTrue(error['text'].startswith(f"Resolve of 'lbry://@some_channel#{some_channel_id[:1]}/bad_content#{bad_content_id[:1]}' was blocked"))
         self.assertTrue(error['censor']['short_url'].startswith('lbry://@blocking#'))
 
         # a filtered/blocked channel impacts all content inside it

